### PR TITLE
color intrinsic types as keywords

### LINKED ISSUE #1809: Well known types should be highlighted as keywords instead of types.
E.g., in the following example, `string` and `unit` are special well known types and should be colorized as keywords (blue) like earlier versions of Visual Studio did.

image

### PR DESCRIPTION
**Pull request by brettfo**
With the change to the Roslyn-based language service, type aliases into the .NET framework are colored as types (teal) instead of keywords (blue).  This is a (probably poor) attempt at fixing that.

This solution is certainly less than ideal, because I wasn't able to find something that I could easily call to determine if a type's token is an alias to a .NET framework type (e.g., `int` -> `System.Int32`).  The closest I could find was [this](https://github.com/Microsoft/visualfsharp/blob/696c6f7e3882caa7b7efcd58b7df70f5a138204d/src/fsharp/NicePrint.fs#L116-L133), but even then, the list wasn't complete; it lacked the `byref` keyword so I had to copy the items into another member function to cover everything.

The diff looks more painful than it really is; I just added one case to the `match` expression:

``` F#
// well known type aliases get colored as keywords 
| CNR(_, (Item.Types (n, _)), _, _, _, _, m) when TypeCheckInfo.IsKeywordType(n) -> 
    Some (m, FSharpTokenColorKind.Keyword) 
```

and the `IsKeywordType()` method itself.

Fixes #1809.

### CODE REVIEW

File: src/fsharp/vs/service.fs [hunk 1]
```diff
@@ -1369,30 +1369,56 @@ type TypeCheckInfo
...
+                     ( ItemOccurence.UseInType
+                     | ItemOccurence.UseInAttribute
+                     | ItemOccurence.Use _
+                     | ItemOccurence.Binding _
+                     | ItemOccurence.Pattern _), _, _, _, m) ->
+                Some (m, FSharpTokenColorKind.TypeName)
+            | _ -> None)
+        |> Seq.toArray
+
+    static member IsKeywordType(name : string) =
```
Comment (KevinRansom) [inline src/fsharp/vs/service.fs]: Should probably include 
````
unit, bigint, 
````

Kevin

Comment (cartermp) [inline src/fsharp/vs/service.fs]: All valid keywords literals are listed here: https://docs.microsoft.com/en-us/dotnet/articles/fsharp/language-reference/literals

Looks like these are also missing in addition to @KevinRansom's suggestion: `int32`, `int64`, `double`, `single`

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

File: src/fsharp/vs/service.fs (hunk 1 above)
Comment (dungpa) [inline src/fsharp/vs/service.fs]: It's better to use `keywordNames` https://github.com/Microsoft/visualfsharp/blob/6afc4f9776a5dba903ca6b11965a206fb8bc23d8/src/fsharp/lexhelp.fsi#L70. Probably convert it to a set and lookup.

**Comment by brettfo** [comment]
Thanks to @cartermp for the pointer to the F# keywords and to @dungpa for suggesting precomputing a set.  I've updated the PR appropriately.

**Comment by brettfo** [comment]
@dotnet-bot test this please

**Pull request merged**